### DISCUSSION TITLE
Setting PathIcon Data from a Binding referencing a StaticResource

### DISCUSSION THREAD

**Discussion by xD0135** [AvaloniaUI/Avalonia#10505] [Q&A] [answered]
Hi, is there a way to do something like this pseudo code:
```
<PathIcon Data="{StaticResource Source={Binding Icon}" />
```

What I'd like to achieve is being able to dynamically load a static resource based on the `Icon` property of my object:
```
public class ListBoxItem
    {
        public string? Icon { get; set; }
        public string? Text { get; set; }
    }
```

I'm able to display the correct icon by hard-coding it, but I'm looking to dynamically set it :
```
<PathIcon Data="{StaticResource grid}" />
```

My ListBox definition is this:
```
                <ListBox Items="{Binding ListBoxItems}" SelectionMode="Single">
                    <ListBox.ItemTemplate>
                        <DataTemplate>
                            <StackPanel Orientation="Horizontal">
                                <PathIcon Data="{Binding Icon}" />
                                <TextBlock Text="{Binding Text}" />
                            </StackPanel>
                        </DataTemplate>
                    </ListBox.ItemTemplate>
                </ListBox>

```
Thank you in advance for your help.

**Comment by timunie** [comment]
You can Bind to Icon and provide a Converter which resolves the Icon for you. You can access resources via App.Current.Resources for example.

**Comment by xD0135** [comment, marked as answer]
Yes, thank you, I ended up doing something similar:

```
        private static StreamGeometry? GetIconForName(string name)
        {
            Application.Current.Resources.TryGetResource(name, out object icon);
            return (StreamGeometry?) icon;
        }
```
Reading https://github.com/AvaloniaUI/Avalonia/issues/6021#issuecomment-1296343550, I agree that we should have a simpler way of accessing app resources through code 😺
